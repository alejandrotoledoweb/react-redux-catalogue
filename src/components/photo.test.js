import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Photo from './photo';

let container = null;

beforeEach(() => {
  container = document.createElement('section');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('it test to render a photo component with id and url of the photo', () => {
  act(() => {
    render(
      <Photo
        id="ZCHj_2lJP00"
        url="https://images.unsplash.com/photo-1592194996308-7b43878e84a6?ixid=MnwyMjE4NTV8MHwxfHNlYXJjaHwxfHxjYXRzfGVufDB8fHx8MTYxODIzNjY2Nw&ixlib=rb-1.2.1"
        photoStyle="photo"
        photoContainer="photo-container"
      />,
      container,
    );
  });

  expect(
    container.innerHTML,
  ).toMatchSnapshot();
});
