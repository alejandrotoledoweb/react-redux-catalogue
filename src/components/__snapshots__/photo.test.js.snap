// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it test to render a photo component with id and url of the photo 1`] = `"<div class=\\"photo-container d-flex align-items-center\\"><img class=\\"photo\\" src=\\"https://images.unsplash.com/photo-1592194996308-7b43878e84a6?ixid=MnwyMjE4NTV8MHwxfHNlYXJjaHwxfHxjYXRzfGVufDB8fHx8MTYxODIzNjY2Nw&amp;ixlib=rb-1.2.1\\" alt=\\"Cat\\"></div>"`;
